indexer/test/test_hca_indexer.py

```diff
@@ -0,0 +1,271 @@
+                old_result_version = result_dict["_source"]["bundles"][0]["version"]
+                old_result_contents = result_dict["_source"]["bundles"][0]["contents"]
+
+                self.assertEqual(self.new_bundle[1], old_result_version)
+                self.assertEqual("Aardvark Ailment", old_result_contents["project"]["project"])
+                self.assertIn("John Denver", old_result_contents["project"]["laboratory"])
+                self.assertIn("Lorem ipsum", old_result_contents["specimens"][0]["species"])
+
+        old_results = get_es_results(self.es_client, self.index_names, check_new_submission)
+
+        mock_index(self.hca_indexer, self.old_bundle, old_data)
+
+        def check_for_overwrite(old_results, new_results):
+            for old_result_dict, new_result_dict in list(zip(old_results, new_results)):
+                old_result_version = old_result_dict["_source"]["bundles"][0]["version"]
+                old_result_contents = old_result_dict["_source"]["bundles"][0]["contents"]
+
+                new_result_version = new_result_dict["_source"]["bundles"][0]["version"]
+                new_result_contents = new_result_dict["_source"]["bundles"][0]["contents"]
+
+                self.assertEqual(old_result_version, new_result_version)
+                self.assertEqual(old_result_contents["project"]["project"],
+                                 new_result_contents["project"]["project"])
+                self.assertEqual(old_result_contents["project"]["laboratory"],
+                                 new_result_contents["project"]["laboratory"])
+                self.assertEqual(old_result_contents["specimens"][0]["species"],
+                                 new_result_contents["specimens"][0]["species"])
+
+        get_es_results(self.es_client, self.index_names, partial(check_for_overwrite, old_results))
+
+    def test_concurrent_specimen_submissions(self):
+        """
+        We submit two different bundles for the same specimen. What happens?
+        """
+        clear_indices(self.es_client, self.index_names)
+        spec1_pack = get_data_by_uuid(self.spec1_bundle[0])
+        spec2_pack = get_data_by_uuid(self.spec2_bundle[0])
+        specimen_list = [(self.spec1_bundle, spec1_pack),
+                         (self.spec2_bundle, spec2_pack)]
+
+        unmocked_mget = Elasticsearch.mget
+
+        def mocked_mget(self, body):
+            mget_return = unmocked_mget(self, body=body)
+            time.sleep(0.5)
+            return mget_return
+
+        def help_index(hca_indexer, specimen_tuple):
+            mock_index(hca_indexer, specimen_tuple[0], specimen_tuple[1])
+
+        with patch.object(Elasticsearch, 'mget', new=mocked_mget):
+            with self.assertLogs(level='WARNING') as cm:
+                with ThreadPoolExecutor(max_workers=2) as executor:
+                    futures_results = executor.map(partial(help_index, self.hca_indexer), specimen_list)
```

Comment:
The var name is misleading. tpe.map() doesn't return futures. It returns an iterable of the return values of the function.

Reply:
You can eliminate `partial` by eliminating the `hca_indexer` parameter and just using `self.hca_indexer` directly in `help_index`.